docs/ocean/design_docs/min_level_cell.rst

```diff
@@ -0,0 +1,459 @@
+Inactive Cells at the Top of the Water Column
+=============================================
+
+date: 2021/02/11
+
+Contributors: Xylar Asay-Davis
+
+
+
+Summary
+-------
+
+For increased flexibility in representing ice-shelf cavities in MPAS-Ocean,
+we require a mechanism for inactivating cells in the water column that are
+above the ice draft (the depth of the ice shelf-ocean interface).  This
+capability would be similar to the disabled cells below the bathymetry.
+Currently, ice-shelf cavities use a vertical coordinate where top layer follows
+the ice draft, but pressure-gradient errors caused by steep, tilted layers
+place many restrictions on this approach (e.g. layers are thick and bathymetry
+is not well represented).  As a step toward a more flexible vertical coordinate
+that can better adapt to changing ice topography in time, we require a
+capability to disable top cells and thus mitigate the constraints imposed by
+the current terrain-following top coordinate.  A successful implementation will
+be able to run an MPAS-Ocean test case with ice-shelf cavities using a
+prescribed vertical coordinate with inactive cells at the top of the water
+column.  The test case will compute "surface" fluxes at the top active layer
+and distribute them from the top down into active layers.  All test cases where
+no top layers are inactive must produce results that remain bit-for-bit
+identical to the current implementation.
+
+
+Requirements
+------------
+
+Requirement: Inactive top cells
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2021/02/11
+
+Contributors: Xylar Asay-Davis
+
+We require a mechanism for indicating that top cells in the water column are
+inactive and preventing computation on these cells.
+
+Requirement: Surface fields and fluxes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Date last modified: 2021/02/11
+
+Contributors: Xylar Asay-Davis
+
+MPAS-Ocean includes several fields that represent the surface values of 3D
+fields.  These fields must use the top active cell in a column rather than the
+first (potentially inactive) cell.  Similarly, the distribution of surface
+fluxes into the water column must begin from the top with the first active
+cell, not the first (potentially inactive) cell.
+
```

Comment:
This requirement is probably too simplistic. The actual requirement is that surface forcing is applied correctly in all cases. This may not be simply making sure it's applied at the top of the active column. When applying the forcing over a depth range, it's not clear what that depth range should be or whether we want the same depth range under ice shelves. Presumably the coupler takes care of part of this by attenuating/masking some fluxes at the ice/ocean interface. But the point is that it's not going to be just a simple shifting of forcing to the top active level and some thought is going to be required for some forcing options. This may also be  true of some vertical mixing parameterizations and options.

Reply:
@philipwjones, thanks for the comment.  I think these concerns would still apply to the terrain-following top coordinate that we currently use.  I agree with you that KPP and other parameterizations might deserve a different treatment under ice shelves, but I think we want to treat that as independent of the top index, and this is something we already have to handle.  So it isn't clear to me how that would affect this design.

Reply:
@xylar for KPP there are a bunch of hard coded 1 indices in the construction of the depth coordinate and surface layer averaging (see here for example - https://github.com/MPAS-Dev/MPAS-Model/blob/ocean/develop/src/core_ocean/shared/mpas_ocn_vmix_cvmix.F#L506-L534).  This could be problematic if the work is just switching loop bounds, but pretty easy to deal with in my view.  Something like
```
do i=1,nEdgesOnCell(iCell)
  iEdge = edgesOnCell(iCell)
  deltaVelocitySquared(minLevelEdge(iEdge))
  ...
  do kIndexOBL = minLevelEdge(iEdge)+1,maxLevelelCell(iCell)
```
So I do think the change is relevant to be aware of here as it is a bit separate from the current terrain following formulation considerations in KPP, but really easy to take care of in implementation.  I think the same would hold for GM routines that have this type of structure.

I don't think we should make too many changes specific to ice shelf cavities, but either way that as you say @xylar is separate from this design.

Reply:
@vanroekel, yes, we definitely plan to make those changes like you are describing (not just loop bounds).  That's fully within the scope of this current work (though KPP is a low priority within that project).

I wanted to have "surface" fields and fluxes as a separate requirement precisely because it's easy to miss them when you're searching for loop bounds.  The KPP example you show is exactly a case like that and what you propose is exactly what I was thinking, too.